pkg/client/entities_client.go

```diff
@@ -14,5 +14,191 @@
 
 package client
 
+import (
+	"encoding/json"
+	"net/url"
+	reflect "reflect"
+	"regexp"
+	"unicode"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/throttle"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/rest"
+)
+
 const pathEntitiesObjects = "/api/v2/entities"
 const pathEntitiesTypes = "/api/v2/entityTypes"
+
+// Full export would be: const defaultListEntitiesFields = "+lastSeenTms,+firstSeenTms,+tags,+managementZones,+toRelationships,+fromRelationships,+icon,+properties"
+// Using smaller export for faster processing, using less memory
+const defaultListEntitiesFields = "+lastSeenTms,+firstSeenTms"
+
+var extraEntitiesFields = map[string][]string{
+	"properties":      {"detectedName", "oneAgentCustomHostName", "ipAddress"},
+	"toRelationships": {"isSiteOf", "isClusterOfHost"},
+}
+
+func getEntitiesTypeFields(entitiesType EntitiesType, ignoreProperties []string) string {
+	typeFields := defaultListEntitiesFields
+
+	for topField, subFieldList := range extraEntitiesFields {
+
+		if contains(ignoreProperties, topField) {
+			continue
+		}
+		fieldSliceObject := getDynamicFieldFromObject(entitiesType, topField)
+
+		if isInvalidReflectionValue(fieldSliceObject) {
+
+		} else {
+			for _, subField := range subFieldList {
+				if contains(ignoreProperties, subField) {
+					continue
+				}
+				if hasSpecificFieldValueInSlice(fieldSliceObject, "id", subField) {
+					typeFields = typeFields + ",+" + topField + "." + subField
+				}
+			}
+		}
+	}
+
+	return typeFields
+}
+
+func contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func getDynamicFieldFromObject(object interface{}, field string) reflect.Value {
+	reflection := reflect.ValueOf(object)
+
+	fieldValue := reflect.Indirect(reflection).FieldByName(field) // nosemgrep: go.lang.security.audit.unsafe-reflect-by-name.unsafe-reflect-by-name
+
+	// We are providing uncapitalized fields from json maps
+	// But GoLang forces capitalized for unmarshalling
+	// Let's try a capitalized first letter
+	if isInvalidReflectionValue(fieldValue) {
+		field = capitalizeFirstLetter(field)
+		// nosemgrep
```

Comment:
this will not do anything on this line. As semgrep scan didn't flag the line above I guess this can be removed

Reply:
that was a mistake, thank you